src/packages/core/extension-registry/models/workspace-action-menu-item.model.ts

```diff
@@ -0,0 +1,22 @@
+import type { ConditionTypes } from '../conditions/types.js';
+import type { MetaEntityAction } from './entity-action.model.js';
+import type { ManifestElementAndApi, ManifestWithDynamicConditions } from '@umbraco-cms/backoffice/extension-api';
+import type { UmbWorkspaceAction } from '@umbraco-cms/backoffice/workspace';
+
+export interface ManifestWorkspaceActionMenuItem
+	extends ManifestElementAndApi<HTMLElement, UmbWorkspaceAction>,
+		ManifestWithDynamicConditions<ConditionTypes> {
+	type: 'workspaceActionMenuItem';
+	meta: MetaWorkspaceActionMenuItem;
+}
+
+export interface MetaWorkspaceActionMenuItem extends MetaEntityAction {
+	/**
+	 * Define which workspace actions this menu item should be shown for.
+	 * @examples [
+	 * 	['Umb.WorkspaceAction.Document.Save', 'Umb.WorkspaceAction.Document.SaveAndPublish']
+	 * ]
+	 * @required
+	 */
+	workspaceActions: string[];
```

Comment:
Do you think it makes sense to mention multiple? My initial though was just one :-) but maybe you have a use case to share?

Reply:
I modeled it after property actions, which take an array of property editor UI aliases. I think it could make sense for something like the "Deploy" button if they want to add the button on both the documents and media workspace. What do you think?